compiler/AST/expr.cpp

```diff
@@ -902,7 +910,10 @@ CallExpr::CallExpr(const char* name,
   argList(),
   partialTag(false),
   methodTag(false),
-  square(false) {
+  square(false),
+  inTry(false),
+  inTryBang(false)
```

Comment:
Would it be useful to have this be set via an enum selector of some kind, since a particular expression is not able to be both a try and a try! expression?

Reply:
I didn't do the enum because it makes it more complex but I could do it & then we'd just have 1 new field. Do you think that would be obviously better?

Reply:
I'm not sure.  This is obviously a complicated issue and I know you've thought about it (likely more than I have).

At first, it seemed like if the outermost `try` was a `try!`, then it wouldn't particularly matter because the line number for the uncaught error would be the same to the user, but then I realized that you would care about any side effects caused by the operations in the other part of the expression (though do we guarantee a particular execution order for them semantically, or is it an implementation detail?).  So if it were true that a `try!` would bleed to the inner parts of the expression, then I think it would simplify things to just have that be the representation once one has been encountered.

And maybe it is true since the implementation here fills in both fields without necessarily denoting which one was most immediately relevant?  I'm still trying to wrap my head around the idea, apologies.

What was your thinking on how an outer `try!` should impact the inner statements?

Reply:
Ugh, that's a little stream of consciousness.

To simplify that, the implementation seems to carry over the context into inner statements, which makes me wonder about the precedence of `try` and `try!` on the inner statements, and whether it would simplify the analysis should one be a stronger statement than the other.